packages/bpk-component-slider/src/BpkSlider.tsx

```diff
@@ -16,60 +17,74 @@
  * limitations under the License.
  */
 
-import PropTypes from 'prop-types';
-
-// @ts-expect-error Untyped import. See `decisions/imports-ts-suppressions.md`.
-import Slider from 'react-slider';
+import * as Slider from '@radix-ui/react-slider';
 
 import { cssModules, isRTL } from '../../bpk-react-utils';
 
 import STYLES from './BpkSlider.module.scss';
 
 const getClassName = cssModules(STYLES);
 
-type Props = {
-  className?: string | null;
+export type Props = {
+  max: number
+  min: number
+  minDistance?: number,
+  step?: number
+  onChange: (value: number[] | number) => void
+  onAfterChange?: (value: number[] | number ) => void
+  value: number[] | number
+  ariaLabel: string[]
+  ariaValuetext?: string[]
   [rest: string]: any;
 }
 
-const BpkSlider = (props: Props) => {
-  const { className, ...rest } = props;
+const BpkSlider = ({ ariaLabel, ariaValuetext, max, min, minDistance, onAfterChange, onChange, step, value, ...rest }: Props) => {
   const invert = isRTL();
-  const classNames = [getClassName('bpk-slider')];
-  const thumbClassNames = [getClassName('bpk-slider__handle')];
-  const trackClassNames = [getClassName('bpk-slider__bar')];
+  const defaultValue = Array.isArray(value) ? value : [value]
+
+  const processSliderValues = (sliderValues: number[], callback?: (val: number | number[]) => void) => {
+  const val = sliderValues.length === 1 ? sliderValues[0] : sliderValues;
+  if (callback) {
+    callback(val);
+   }
+  };
+
+  const handleOnChange = (sliderValues: number[]) => {
+    processSliderValues(sliderValues, onChange);
+  };
 
-  const isRange = (rest.value || rest.defaultValue || []).length > 1;
+  const handleOnAfterChange = (sliderValues: number[]) => {
+    processSliderValues(sliderValues, onAfterChange);
+  };
```

Comment:
This looks a bit wild but its to handle differences between the base libraries. This new library always returns the value in an array, so if there is just one thumb it is passed back like `[value]` where as for a range its the expected `[value, value]`

The previous library returned a one thumb slider value back as just a number rather than a number in an array

Reply:
Yea, that is from the internal workings of the library. Somewhere in here, its state is an array

https://github.com/radix-ui/primitives/blob/main/packages/react/slider/src/Slider.tsx#L99